js/main.js

```diff
@@ -114,61 +140,60 @@
 
       }
     ])
-    .service('true', ['$q',
-      function (q) {
-        return {
-          'get': function () {
-            var deferred = q.defer();
-
-            deferred.resolve(true);
+    .service('mockInterruptService', ['$q', '$log',
+      function (q, log) {
 
-            return deferred.promise;
-          }
+        function promise(value) {
+          var deferred = q.defer();
+          deferred.resolve(value);
+          return deferred.promise;
         }
-      }
-    ])
-    .service('false', ['$q',
-      function (q) {
+
         return {
-          'get': function () {
-            var deferred = q.defer();
 
-            deferred.resolve(false);
+          isInterruptRequired: function (interruptType) {
+            if (location.search.indexOf(interruptType) >= 0) {
+              return promise(true);
+            } else {
+              return promise(false);
```

Comment:
Should this reject the promise on false instead of resolving with a false value? Either way works, and you are checking in `buildInterruptStep` for the true/false.